src/dotnet/UI.Blazor.App/Components/PlaceInfo/PlaceInfoPage.razor

```diff
@@ -1,61 +1,71 @@
+    <chat-view-skeleton count="15" />
+    return;
+}
+@if (ReferenceEquals(place, SpecialPlace.NoPlaceSelected)) {
+    <div class="no-chat-body">
+        Select a place.
     </div>
-</RenderIntoSlot>
-
-<PlaceInfo PlaceId="@m.SelectedPlaceId"/>
+} else if (ReferenceEquals(place, SpecialPlace.Unavailable)) {
+    <ChatNotFound ShowSignIn="@false" Target="@ChatNotFound.TargetKind.Place" />
+} else {
+    <PlaceInfo PlaceId="@m.Place.Id" />
+}
 
 @code {
-    private PlaceId _selectedPlaceId = PlaceId.None;
+    private ILogger? _log;
+    private Model? _rendered;
 
-    [Inject] private ChatUIHub Hub { get; init; } = null!;
+    [Inject] protected ChatUIHub Hub { get; init; } = null!;
+    protected Session Session => Hub.Session();
+    protected IPlaces Places => Hub.Places;
+    protected ILogger Log => _log ??= Hub.LogFor(GetType());
 
-    private Session Session => Hub.Session();
-    private IPlaces Places => Hub.Places;
+    [Parameter] public string? PlaceSid { get; set; }
 
     protected override ComputedState<Model>.Options GetStateOptions()
-        => ComputedStateComponent.GetStateOptions(GetType(),
-            static t => new ComputedState<Model>.Options() {
-                InitialValue = new Model(),
-                Category = ComputedStateComponent.GetStateCategory(t),
-            });
+        => new() {
+            InitialValue = !PlaceSid.IsNullOrEmpty() ? Model.Loading : new Model(SpecialPlace.NoPlaceSelected),
+            Category = ComputedStateComponent.GetStateCategory(GetType()),
+        };
 
     protected override async Task<Model> ComputeState(CancellationToken cancellationToken) {
-        var navbarSettings = await Hub.ChatUI.NavbarSettings.Use(cancellationToken).ConfigureAwait(false);
-        var placeIds = await Hub.Contacts.ListPlaceIds(Session, cancellationToken).ConfigureAwait(false);
-
-        var maybePlaces = await placeIds
-            .OrderBy(x => navbarSettings.PlacesOrder.IndexOf(x))
-            .Select(id => Hub.Places.Get(Session, id, cancellationToken))
-            .Collect(ApiConstants.Concurrency.Unlimited, cancellationToken)
-            .ConfigureAwait(false);
-        var places = maybePlaces.SkipNullItems().ToList();
-        if (_selectedPlaceId == PlaceId.None && places.Count > 0)
-            _selectedPlaceId = places[0].Id;
-        return new() {
-            Places = places,
-            SelectedPlaceId = _selectedPlaceId,
-        };
-    }
+        var placeSid = PlaceSid;
+
+        if (placeSid.IsNullOrEmpty())
+            return new Model(SpecialPlace.NoPlaceSelected);
 
-    private void OnPlaceClick(PlaceId placeId) {
-        if (placeId == PlaceId.None)
-            return;
-        _selectedPlaceId = placeId;
-        State.Recompute();
+        if (!PlaceId.TryParse(placeSid, out var placeId))
+            return Model.Unavailable;
+
+        try {
+            var place = await Places.Get(Session, placeId, cancellationToken).ConfigureAwait(false);
```

Comment:
needs extract to local variable

Reply:
I don't get what should be extracted to local var?